api/src/services/listing.service.ts

```diff
@@ -655,9 +689,32 @@ export class ListingService implements OnModuleInit {
       },
     );
 
+    if (dto.units?.length > 0 && dto.unitGroups?.length > 0) {
+      throw new BadRequestException({
+        message: 'Cannot provide both units and unitGroups',
+        status: 400,
+      });
+    }
+
+    const rawJurisdiction = await this.prisma.jurisdictions.findFirst({
+      where: {
+        id: dto.jurisdictions.id,
+      },
+      include: {
+        featureFlags: true,
+      },
+    });
+
+    const enableUnitGroups = rawJurisdiction?.featureFlags?.find(
+      (featureFlag) => featureFlag.name === FeatureFlagEnum.enableUnitGroups,
+    )?.active;
+
     dto.unitsAvailable =
-      dto.reviewOrderType !== ReviewOrderTypeEnum.waitlist && dto.units
-        ? dto.units.length
+      dto.reviewOrderType !== ReviewOrderTypeEnum.waitlist &&
+      (enableUnitGroups ? dto?.unitGroups?.length : dto?.units?.length)
+        ? enableUnitGroups
+          ? dto?.unitGroups?.length
+          : dto?.units?.length
```

Comment:
I'm quite sure it's not right 😞 . I don't even know if we're using that in Detroit right now. in core it seem to be the one in partners listing table. So i don't know if i should just make it `undefined` for `enableUnitGroups`, or sum `totalAvailable` from each `unitGroup` 🤔
@ludtkemorgan